### PR TITLE
RESTWS-660: SessionController to GET and POST locale information.

### PR DESCRIPTION
**Pull request by mks-d**

### CODE REVIEW

**Comment by dkayiwa** [comment]
The ticket status says you are still doing more work on this. 😊

**Comment by mks-d** [comment]
Sorry... just transitioned it.

File: omod-1.8/src/test/java/org/openmrs/module/webservices/rest/web/v1_0/controller/openmrs1_8/SessionController1_8Test.java [hunk 1]
```diff
@@ -77,4 +96,26 @@ public void get_shouldReturnTheSessionIdIfTheUserIsNotAuthenticated() throws Exc
...
+	}
+	
+	@Test(expected = RuntimeException.class)
+	public void post_shouldFailWhenSettingIllegalLocale() throws Exception {
+		String newLocale = "wx_YZ";
+		String content = "{\"locale\":\"" + newLocale + "\"}";
+		controller.post(new ObjectMapper().readValue(content, HashMap.class));
+	}
+	
+	@Test(expected = RuntimeException.class)
```
Comment (dkayiwa) [inline omod-1.8/src/test/java/org/openmrs/module/webservices/rest/web/v1_0/controller/openmrs1_8/SessionController1_8Test.java]: Shouldn't this be APIException?

File: omod-1.8/src/test/java/org/openmrs/module/webservices/rest/web/v1_0/controller/openmrs1_8/SessionController1_8Test.java [hunk 2]
```diff
@@ -77,4 +96,26 @@ public void get_shouldReturnTheSessionIdIfTheUserIsNotAuthenticated() throws Exc
...
+	
+	@Test
+	public void post_shouldSetTheUserLocale() throws Exception {
+		Locale newLocale = new Locale("sp");
+		String content = "{\"locale\":\"" + newLocale.toString() + "\"}";
+		controller.post(new ObjectMapper().readValue(content, HashMap.class));
+		Assert.assertEquals(newLocale, Context.getLocale());
+	}
+	
+	@Test(expected = RuntimeException.class)
```
Comment (dkayiwa) [inline omod-1.8/src/test/java/org/openmrs/module/webservices/rest/web/v1_0/controller/openmrs1_8/SessionController1_8Test.java]: Shouldn't this be APIException?

File: omod-1.8/src/test/java/org/openmrs/module/webservices/rest/web/v1_0/controller/openmrs1_8/SessionController1_8Test.java [hunk 3]
```diff
@@ -77,4 +96,26 @@ public void get_shouldReturnTheSessionIdIfTheUserIsNotAuthenticated() throws Exc
 		Assert.assertEquals(SESSION_ID, PropertyUtils.getProperty(ret, "sessionId"));
 		Assert.assertEquals(false, PropertyUtils.getProperty(ret, "authenticated"));
 	}
+	
+	@Test
+	public void post_shouldSetTheUserLocale() throws Exception {
+		Locale newLocale = new Locale("sp");
+		String content = "{\"locale\":\"" + newLocale.toString() + "\"}";
+		controller.post(new ObjectMapper().readValue(content, HashMap.class));
+		Assert.assertEquals(newLocale, Context.getLocale());
```
Comment (dkayiwa) [inline omod-1.8/src/test/java/org/openmrs/module/webservices/rest/web/v1_0/controller/openmrs1_8/SessionController1_8Test.java]: Can you add Assert.assertNotEquals(newLocale, Context.getLocale()) before posting?

**Comment by mks-d** [comment]
@dkayiwa, I have made changes according to your recommendations.

**Pull request merged**